applications/RomApplication/custom_utilities/rom_auxiliary_utilities.cpp

```diff
@@ -0,0 +1,440 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "containers/pointer_vector_set.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+
+// Application includes
+#include "rom_application_variables.h"
+#include "rom_auxiliary_utilities.h"
+
+namespace Kratos
+{
+
+namespace
+{
+    std::map<GeometryData::KratosGeometryType, std::string> AuxiliaryGeometryToConditionMap {
+        {GeometryData::KratosGeometryType::Kratos_Line2D2, "LineCondition2D2N"},
+        {GeometryData::KratosGeometryType::Kratos_Triangle3D3, "SurfaceCondition3D3N"},
+        {GeometryData::KratosGeometryType::Kratos_Triangle3D6, "SurfaceCondition3D6N"},
+        {GeometryData::KratosGeometryType::Kratos_Quadrilateral3D4, "SurfaceCondition3D4N"},
+        {GeometryData::KratosGeometryType::Kratos_Quadrilateral3D8, "SurfaceCondition3D8N"},
+        {GeometryData::KratosGeometryType::Kratos_Quadrilateral3D9, "SurfaceCondition3D9N"},
+    };
+}
+
+void RomAuxiliaryUtilities::SetHRomComputingModelPart(
+    const Parameters HRomWeights,
+    const ModelPart& rOriginModelPart,
+    ModelPart& rHRomComputingModelPart)
+{
+    // Ensure that the provided destination model part is empty
+    rHRomComputingModelPart.Clear();
+
+    // Auxiliary containers to save the entities involved in the HROM mesh
+    // Note that we use a set for the nodes to make sure that the same node is not added by more than one element/condition
+    NodesPointerSetType hrom_nodes_set;
+    std::vector<Element::Pointer> hrom_elems_vect;
+    std::vector<Condition::Pointer> hrom_conds_vect;
+
+    const auto& r_elem_weights = HRomWeights["Elements"];
+    hrom_elems_vect.reserve(rOriginModelPart.NumberOfElements());
+    for (auto it = r_elem_weights.begin(); it != r_elem_weights.end(); ++it) {
+        // Get element from origin mesh
+        const IndexType elem_id = stoi(it.name());
+        const auto p_elem = rOriginModelPart.pGetElement(elem_id + 1); //FIXME: WHY THIS +1?
```

Comment:
Is it going to remain +1?  For me, it is ok for now...

Reply:
I'd change the Ids 0 issue in a standalone PR to easily backtrack the changes if something gets messed up.